solr/core/src/java/org/apache/solr/schema/IndexSchema.java

```diff
@@ -138,7 +138,7 @@
 
   protected List<SchemaField> fieldsWithDefaultValue = new ArrayList<>();
   protected Collection<SchemaField> requiredFields = new HashSet<>();
-  protected volatile DynamicField[] dynamicFields;
```

Comment:
@sarowe why was this volatile?  It's fishy to see this as the only volatile field.

Reply:
> @sarowe why was this volatile? It's fishy to see this as the only volatile field.

I was looking for any double check pattern, or lazy init, found nothing, as you mentioned it is the only one, so i removed it, makes no sense from my point of view